Riot/Modules/TabBar/TabBarCoordinator.swift

```diff
@@ -705,6 +721,128 @@ final class TabBarCoordinator: NSObject, TabBarCoordinatorType {
+extension TabBarCoordinator {
+    private func promptUserToFillAnEmailToInvite(completion: @escaping ((String) -> Void)) {
+        currentAlertController?.dismiss(animated: false)
+        
+        let alertController = UIAlertController(title: TchapL10n.contactsInviteByEmailTitle,
+                                                message: TchapL10n.contactsInviteByEmailMessage,
+                                                preferredStyle: .alert)
+        
+        // Add textField
+        alertController.addTextField(configurationHandler: { textField in
+            textField.isSecureTextEntry = false
+            textField.placeholder = nil
+            textField.keyboardType = .emailAddress
+        })
+        
+        // Cancel action
+        let cancelAction = UIAlertAction(title: TchapL10n.actionCancel,
+                                         style: .cancel) { [weak self] _ in
+            self?.currentAlertController = nil
+        }
+        alertController.addAction(cancelAction)
+        
+        // Invite action
+        let inviteAction = UIAlertAction(title: TchapL10n.actionInvite,
+                                         style: .default) { [weak self] _ in
+            guard let currentAlertController = self?.currentAlertController,
+                  let email = currentAlertController.textFields?.first?.text?.lowercased() else {
+                return // FIXME: Verify if dismiss should be needed in this case
+            }
+            
+            self?.currentAlertController = nil
+            
+            if MXTools.isEmailAddress(email) {
+                completion(email)
+            } else {
+                self?.currentAlertController?.dismiss(animated: false)
+                let errorAlertController = UIAlertController(title: TchapL10n.authenticationErrorInvalidEmail,
+                                                             message: nil,
+                                                             preferredStyle: .alert)
+                let okAction = UIAlertAction(title: VectorL10n.ok,
+                                             style: .default) { [weak self] _ in
+                    self?.currentAlertController = nil
+                }
+                errorAlertController.addAction(okAction)
+                errorAlertController.mxk_setAccessibilityIdentifier("ContactsVCInviteByEmailError")
+                self?.currentAlertController = errorAlertController
+                self?.masterTabBarController.present(errorAlertController, animated: true)
+            }
+        }
+        alertController.addAction(inviteAction)
+        alertController.mxk_setAccessibilityIdentifier("ContactsVCInviteByEmailDialog")
+
+        self.currentAlertController = alertController
+        
+        masterTabBarController.present(alertController, animated: true)
+    }
+    
+    private func sendEmailInvite(to email: String) {
+        guard let inviteService = self.inviteService else { return }
```

Comment:
There is a pb here
During my test no invite was sent because this inviteService was nil
An error would be useful

I think inviteService should be created here if nil, and an error should be display if the new creation failed
A race condition prevent it from being created on init (I think)

Reply:
It was not nil on my side, but to fix this, I moved the inviteService instantiation into the `sendEmailInvite` method in the last commit.